py/server/deephaven/arrow.py

```diff
@@ -0,0 +1,148 @@
+#
+#     Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+"""This module supports conversions between Arrow tables and Deephaven tables."""
+
+from typing import List, Dict
+
+import jpy
+import pyarrow as pa
+
+from deephaven import DHError
+from deephaven.table import Table
+
+_JArrowToTableConverter = jpy.get_type("io.deephaven.extensions.barrage.util.ArrowToTableConverter")
+_JTableToArrowConverter = jpy.get_type("io.deephaven.extensions.barrage.util.TableToArrowConverter")
+
+_ARROW_DH_DATA_TYPE_MAPPING = {
+    pa.null(): '',
+    pa.bool_(): 'java.lang.Boolean',
+    pa.int8(): 'byte',
+    pa.int16(): 'short',
+    pa.int32(): 'int',
+    pa.int64(): 'long',
+    pa.uint8(): '',
+    pa.uint16(): 'char',
+    pa.uint32(): '',
+    pa.uint64(): '',
+    pa.float16(): '',
+    pa.float32(): 'float',
+    pa.float64(): 'double',
+    pa.time32('s'): '',
+    pa.time32('ms'): '',
+    pa.time64('us'): '',
```

Comment:
Any reason these can't be mapped to DateTime?  Are we only supporting types that have full round-trip?  Are we not supporting type mapping?  etc.

Reply:
@nbauernfeind?

Reply:
We can, but this requires a separate set of java changes and should not be bundled here.

Currently, we only support types that can be round-tripped and have a unique on-to mapping. To support multiple source types mapping to the same destination type requires java-parsing-side effort.

Reply:
This looks like it is covered in https://github.com/deephaven/deephaven-core/issues/3223